rules/ruleset.go

```diff
@@ -956,31 +967,31 @@ func (rs *ruleSet) getRollupRuleByID(id string) (*rollupRule, error) {
 	return nil, errNoSuchRule
 }
 
-func (rs *ruleSet) latestMappingRules() ([]*MappingRuleView, error) {
+func (rs *ruleSet) latestMappingRules() (map[string]*MappingRuleView, error) {
 	mrs, err := rs.MappingRules()
 	if err != nil {
 		return nil, err
 	}
-	result := make([]*MappingRuleView, 0, len(mrs))
+	result := make(map[string]*MappingRuleView, len(mrs))
 	for _, m := range mrs {
 		if len(m) > 0 && !m[0].Tombstoned {
 			// views included in m are sorted latest first.
-			result = append(result, m[0])
+			result[m[0].ID] = m[0]
```

Comment:
rs.MappingRules returns a map. Use the key instead of m[0].ID

Reply:
Same for the rollup

Reply:
Sure

Reply:
Done